config/v1/types_infrastructure.go

```diff
@@ -157,12 +209,45 @@ type PlatformStatus struct {
 	IBMCloud *IBMCloudPlatformStatus `json:"ibmcloud,omitempty"`
 }
 
+// AWSServiceEndpoint store the configuration of a custom url to
+// override existing defaults of AWS Services.
+type AWSServiceEndpoint struct {
+	// Name is the name of the AWS service.
+	// This must be provided and cannot be empty.
+	Name string `json:"name"`
+
+	// URL is hostname only or fully qualified URI that overrides the default generated
+	// endpoint for a client.
+	// This must be provided and cannot be empty.
+	URL string `json:"url"`
```

Comment:
@damemi do we default to required or optional?  Is there any downside to simply specifying everything?